pkg/resources/cruisecontrol/configmap.go

```diff
@@ -198,13 +231,47 @@ func GenerateCapacityConfig(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger,
 
 		capacityConfig.BrokerCapacities = append(capacityConfig.BrokerCapacities, brokerCapacity)
 	}
+	return capacityConfig.BrokerCapacities, nil
+}
 
-	result, err := json.MarshalIndent(capacityConfig, "", "    ")
+func extendUserProvidedCapacityConfig(data []byte, kafkaCluster *v1beta1.KafkaCluster, log logr.Logger) (string, error) {
```

Comment:
I don't agree really with either of these, this function is only used when extending user provided config, hence the name. Sure I can use something more generic but what's the point until it's not needed elsewhere? Also I think the only thing that would change by passing in `*JBODInvariantCapacityConfig` instead of `[]byte` is that the unmarshal would have to happen before calling this function. I might be wrong on the second one though.

Reply:
Had an offline discussion about this, I understand now what you meant. I'll change it up a bit.